example-apps/internal-knowledge-search/app-ui/src/pages/SearchPage.tsx

```diff
@@ -0,0 +1,111 @@
+import SearchBar from "../components/SearchBar";
+import FilterBar from "../components/FilterBar";
+import SearchResults from "../components/SearchResults";
+import React, {useEffect, useState} from "react";
+
+import SearchApplicationClient from '@elastic/search-application-client';
+import {useDispatch, useSelector} from "react-redux";
+import {useToast} from "../contexts/ToastContext";
+import {updateSearchResults} from "../store/slices/searchResultsSlice";
+import SortBar from "../components/SortBar";
+import {MessageType} from "../components/Toast";
+import dataSourceToLogoPathLookup from '../config/dataSourceToLogoPathLookup.json';
+import documentsToSearchResultsMappings from '../config/documentsToSearchResultMappings.json';
+import {RootState} from "../store/store";
+
+
+const mapHitToSearchResult = (hit) => {
+    const doc = hit._source;
+    const dataSource = hit._index;
+    const dataSourceImagePath = dataSource in dataSourceToLogoPathLookup ? dataSourceToLogoPathLookup[dataSource] : "";
+    const dataSourceMapping = documentsToSearchResultsMappings[dataSource]
+
+    return {
+        title: doc[dataSourceMapping["title"]] || "No title",
+        created: doc[dataSourceMapping["created"]] || "No creation date",
+        previewText: doc[dataSourceMapping["previewText"]] || "No preview",
+        fullText: doc[dataSourceMapping["fullText"]] || "No full text",
+        dataSource: dataSource || "Data source unknown",
+        dataSourceImagePath: dataSourceImagePath || "",
+        link: doc[dataSourceMapping["link"]] || "Unknown"
+    };
+}
+
+export default function SearchPage() {
+    const dispatch = useDispatch();
+    const {showToast} = useToast();
+    const [loading, setLoading] = useState(false);
+    //TODO: simplify query state and move it to one place
+    const [query, setQuery] = useState<string>("");
+
+    const {appName, apiKey, searchEndpoint} = useSelector((state: RootState) => state.searchApplicationSettings);
+    const {sorts} = useSelector((state: RootState) => state.sort);
+    const indexFilter = useSelector((state: RootState) => state.filter)["filters"]["Data sources"].values;
+
+
+    useEffect(() => {
+        const fetchData = async () => await handleSearchSubmit();
```

Comment:
No big deal, but you can simply call handleSearchSubmit() is doesn't make sense to wrap it and call